libevmasm/RuleList.h

```diff
@@ -507,6 +507,18 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		});
 	}
 
+	// Move AND with constant inside OR
+	rules.push_back({
+		// AND(OR(X, Y), A) -> OR(AND(X, A), AND(Y, A))
+		Builtins::AND(Builtins::OR(X, Y), A),
+		[=]() -> Pattern { return Builtins::OR(Builtins::AND(X, A), Builtins::AND(Y, A)); }
+	});
+	rules.push_back({
+		// AND(A, OR(X, Y)) -> OR(AND(X, A), AND(Y, A))
```

Comment:
Call me silly, but should we add a python proof of the commuted one as well, just for completeness sake :-)?

Reply:
Added.